src/java.base/share/classes/sun/security/x509/X500Name.java

```diff
@@ -416,17 +418,9 @@ public boolean equals(Object obj) {
             return this.canonicalDn.equals(other.canonicalDn);
         }
         // quick check that number of RDNs and AVAs match before canonicalizing
-        int n = this.names.length;
-        if (n != other.names.length) {
+        if (!Arrays.equals(this.names, other.names,
+                Comparator.comparingInt(n -> n.assertion.length)))
```

Comment:
I'd keep the original comparison of the lengths; its a lot less magical than `Comparator.comparingInt(n -> n.assertion.length))`

Reply:
I tend to agree with Roger. I have to think more about what is being compared with the new code.

Reply:
Yeah, I assume no matter how one shapes it, it doesn't seem a readability win:

	if (!Arrays.equals(this.names, other.names,
	        (n1, n2) -> Integer.compare(n1.assertion.length, n2.assertion.length))) {
	    return false;
	}

Reverted in 42d9543b769.